res/cmake/def.cmake

```diff
@@ -8,9 +8,26 @@ else() # !Clang
   set (PHARE_FLAGS ${PHARE_FLAGS} --param=min-pagesize=0 )
 endif() # clang
 
+set (PHARE_LINK_FLAGS )
+set (PHARE_BASE_LIBS )
+
+if(PGO_GEN)
```

Comment:
there's no automated mechanism in cmake to achieve this that I know if.
as such, we should probably have some checks that we are using a compiler that supports these options